compiler/rustc_codegen_ssa/src/mir/debuginfo.rs

```diff
@@ -438,11 +438,10 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                 let (var_ty, var_kind) = match var.value {
                     mir::VarDebugInfoContents::Place(place) => {
                         let var_ty = self.monomorphized_place_ty(place.as_ref());
-                        let var_kind = if self.mir.local_kind(place.local) == mir::LocalKind::Arg
+                        let var_kind = if let Some(arg_index) = var.argument_index
                             && place.projection.is_empty()
-                            && var.source_info.scope == mir::OUTERMOST_SOURCE_SCOPE
                         {
-                            let arg_index = place.local.index() - 1;
+                            let arg_index = arg_index as usize - 1;
```

Comment:
This is definitely pre-existing, but the only uses of `arg_index` below add 1 back to the value.
It might make sense to remove this juggling, but possibly also to document that `VariableKind:: ArgumentVariable` is also 1-indexed. (And possibly check that this is in fact the case where `VariableKind` is used.)

Reply:
Good idea. I didn't do that originally because I wasn't sure what to do with eddyb's TODO questioning whether the +1 was actually necessary. However, I've now tried emitting argument indexes starting from 0 and it causes gdb to not show argument 0, so I guess that answers that question - we do need to start from 1. I've removed that part of the TODO and removed the `-1 and +1` juggling.